node/core/approval-voting/src/lib.rs

```diff
@@ -106,6 +107,28 @@ enum Mode {
 	Syncing(Box<dyn SyncOracle + Send>),
 }
 
+enum TaskStatus {
+	InProgress,
+	Executed(ValidationResult),
+}
+
+struct ApprovalState {
+	block_hash: Hash,
+	status: TaskStatus,
+}
+
+impl ApprovalState {
+	fn new(block_hash: Hash) -> Self {
+		Self { block_hash, status: TaskStatus::InProgress }
+	}
+
+	fn verified(block_hash: Hash, validation_result: ValidationResult) -> Self {
+		Self { block_hash, status: TaskStatus::Executed(validation_result) }
+	}
+}
+
+type ApprovalsCache = Arc<Mutex<lru::LruCache<CandidateHash, ApprovalState>>>;
```

Comment:
I needed to pass pass the inner LruCache as mutable, so it is insufficient to use an Arc alone. I opted to introduce a `std::sync::Mutex`, but I'm not super happy with needing to use a Mutex here. Is there a better way?